src/components/media/MediasSelectModal.jsx

```diff
@@ -150,41 +195,63 @@ const MediasSelectModal = ({
                   ))}
                 </div>
               </div>
-              {/* Media */}
-              {filteredMedias.map((media, mediaItemIndex) => (
-                <MediaCard
-                  type={mediaRoom}
-                  media={media}
-                  mediaItemIndex={mediaItemIndex}
-                  onClick={() => onMediaSelect(media)}
-                  key={media.name}
-                  isSelected={media.name === watch("selectedMedia")?.name}
-                  showSettings={false}
+            </Skeleton>
+
+            <Skeleton
+              w="100%"
+              h={isListMediaFilesLoading ? "4.5rem" : "fit-content"}
+              isLoaded={!isListMediaFilesLoading}
+            >
+              <SimpleGrid w="100%" columns={3} spacing="2.5%">
+                {files
+                  .filter(({ data }) => filteredMedias.includes(data?.name))
+                  .map(({ data, isLoading }, mediaItemIndex) => (
+                    <Skeleton
+                      w="100%"
+                      h={isLoading ? "4.5rem" : "fit-content"}
+                      isLoaded={!isLoading}
+                    >
+                      <MediaCard
+                        type={mediaRoom}
+                        media={data}
+                        mediaItemIndex={mediaItemIndex}
+                        onClick={() => onMediaSelect(data)}
+                        key={data.name}
```

Comment:
The key should be set on the `<Skeleton>` tag instead.

Reply:
not entirely sure why here, could you explain this to me?

Reply:
Because each array item [needs to have a unique key](https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key) that can uniquely identify it among other items, and in this case the `<Skeleton>` tag is now the one used to compare siblings, instead of `<MediaCard>` which was originally the case.